IdentityCore/src/requests/sdk/msal/MSIDDefaultBrokerTokenRequest.m

```diff
@@ -22,20 +22,34 @@
 // THE SOFTWARE.
 
 #import "MSIDDefaultBrokerTokenRequest.h"
+#import "MSIDInteractiveRequestParameters.h"
+#import "MSIDAccountIdentifier.h"
 
 @implementation MSIDDefaultBrokerTokenRequest
 
 // Those parameters will be different depending on the broker protocol version
 - (NSDictionary *)protocolPayloadContentsWithError:(NSError **)error
 {
-    // TODO: MSAL pieces
-    return @{};
+    NSString *homeAccountId = self.requestParameters.accountIdentifier.homeAccountId;
+    NSString *username = self.requestParameters.accountIdentifier.legacyAccountId;
+    
+    NSDictionary *contents =
+    @{
+      @"scope": self.requestParameters.target ?: @"",
+      @"home_account_id": homeAccountId ?: @"",
+      @"username": username ?: @"",
```

Comment:
any reasons on why we send it for empty string? I think we should not send it for them

Reply:
fixed. Now only send if value is not empty :)